added firestore using a note taking demo

mysticetii opened the pull request.

joykuotw commented on lib/firebase_firestore/firestore_page.dart:
```diff
@@ -0,0 +1,152 @@
+      if (isUserLoggedIn) {
+        DocumentSnapshot doc = await usersNote.get();
+        Map<String, dynamic> userData = doc.data();
+        setState(() {
+          _editingController.value = TextEditingValue(text: userData['note']);
+          isInEditingMode = userData['editMode'];
+        });
```
you should handle if userData is null, or it will cause an exception.

mysticetii replied:
Thank you for pointing that out. I have added check in the next commit.

joykuotw commented on lib/firebase_firestore/firestore_page.dart:
```diff
@@ -0,0 +1,152 @@
+
+    Widget lastRow() {
+      Widget rowWidget;
+      rowWidget = (_auth.currentUser != null)
```
it would be better to always display the save button and indicate user to login to save note.

mysticetii replied:
Thank you for the suggestion. I will be modifying it in the next commit.

mysticetii replied:
The navigation has been updated.

joykuotw commented on lib/firebase_firestore/firestore_page.dart:
```diff
@@ -65,56 +77,42 @@ class _FirestorePageState extends State<FirestorePage> {
             .then((value) => print("Note Added"))
             .catchError((error) => print("Failed to add note: $error"));
+      } else {
+        _pageNavigator.setFromIndex = _pageNavigator.getCurrentPageIndex;
```
redundant, already set at line 48.

mysticetii replied:
Thank you for pointing that out. I have removed the redundant code

joykuotw reviewed (changes requested):
1. If you have already loggined and use side drawer to log out at Firestore page, the save button doesn't prompt to login when clicked at this point.
2. Need to handle empty email or password during login, or the app will crash.

mysticetii commented:
I have added additional checks for the empty email + password field. Before the firebase auth was handling the null field. Updated the auth provider so that if users log out from the drawer window the save button in firestore demo will prompt for login again.

joykuotw commented:
It works but would be better to just use a listener to `authStateChanges()` in the Auth provider and update the user login status, so you don't need to use the `StreamBuilder` and can get the things you want from the provider instead.

Merge it and I will make the update later then.
